holidays/countries/australia.py

```diff
@@ -53,227 +64,317 @@ def _populate(self, year):
         # TODO do more research on history of Aus holidays
 
         # New Year's Day
-        name = "New Year's Day"
-        jan1 = date(year, JAN, 1)
-        self[jan1] = name
-        if self.observed and self._is_weekend(jan1):
-            self[_get_nth_weekday_from(1, MON, jan1)] = name + " (Observed)"
+        jan1 = self._add_new_years_day("New Year's Day")
+        if self._is_weekend(jan1):
+            self._add_observed(jan1)
 
         # Australia Day
-        jan26 = date(year, JAN, 26)
         if year >= 1935:
-            if self.subdiv == "NSW" and year < 1946:
-                name = "Anniversary Day"
-            else:
-                name = "Australia Day"
-            self[jan26] = name
-            if self.observed and year >= 1946 and self._is_weekend(jan26):
-                self[_get_nth_weekday_from(1, MON, jan26)] = (
-                    name + " (Observed)"
-                )
+            name = (
+                "Anniversary Day"
+                if self.subdiv == "NSW" and year <= 1945
+                else "Australia Day"
+            )
+            jan26 = self._add_holiday(name, JAN, 26)
+            if year >= 1946 and self._is_weekend(jan26):
+                self._add_observed(jan26)
         elif year >= 1888 and self.subdiv != "SA":
-            name = "Anniversary Day"
-            self[jan26] = name
+            self._add_holiday("Anniversary Day", JAN, 26)
```

Comment:
You can override `_add_subdiv_holidays()` and extract this [non-]`SA` case there.
I guess I've only used it for Portugal so far.